.github/actions/cherry-pick/action.yml

```
@@ -0,0 +1,206 @@
+        PR_TITLE="${{ github.event.pull_request.title }}"
+        PR_AUTHOR="${{ github.event.pull_request.user.login }}"
+
+        echo "Processing PR #$PR_NUMBER (reason: ${{ steps.should_run.outputs.reason }})"
+
+        # Determine which labels to process
+        if [ "${{ steps.should_run.outputs.reason }}" = "label_added_to_merged_pr" ]; then
+          # Only process the specific label that was just added
+          LABEL_NAME="${{ github.event.label.name }}"
+          if [[ "$LABEL_NAME" =~ ^backport-(.+)$ ]]; then
+            LABELS="$LABEL_NAME"
+          else
+            echo "Label $LABEL_NAME does not match backport pattern"
+            exit 0
+          fi
+        else
+          # PR was just merged, process all backport labels
+          LABELS=$(gh pr view $PR_NUMBER --json labels --jq '.labels[].name' | grep '^backport-' || true)
+        fi
+
+          if [ -z "$LABELS" ]; then
+            echo "No backport labels found"
+            exit 0
+          fi
+
+          echo "Found backport labels: $LABELS"
+
+          # Process each backport label
+          echo "$LABELS" | while read -r label; do
+            if [[ "$label" =~ ^backport-(.+)$ ]]; then
+              TARGET_BRANCH="${BASH_REMATCH[1]}"
+              echo "Processing backport to branch: $TARGET_BRANCH"
+
+              # Check if target branch exists
+              if ! git ls-remote --heads origin "$TARGET_BRANCH" | grep -q "$TARGET_BRANCH"; then
+                echo "❌ Target branch $TARGET_BRANCH does not exist, skipping"
+
+                # Comment on the original PR about the missing branch
+                gh pr comment $PR_NUMBER --body "⚠️ Cannot backport to \`$TARGET_BRANCH\`: branch does not exist."
+                continue
+              fi
+
+              # Create a unique branch name for the cherry-pick
+              CHERRY_PICK_BRANCH="cherry-pick-${PR_NUMBER}-to-${TARGET_BRANCH}"
+
+              # Check if a cherry-pick PR already exists
+              EXISTING_PR=$(gh pr list --head "$CHERRY_PICK_BRANCH" --json number --jq '.[0].number' 2>/dev/null || echo "")
+              if [ -n "$EXISTING_PR" ]; then
+                echo "⚠️ Cherry-pick PR already exists: #$EXISTING_PR"
+                gh pr comment $PR_NUMBER --body "Cherry-pick to \`$TARGET_BRANCH\` already exists: #$EXISTING_PR"
+                continue
+              fi
+
+              # Fetch and checkout target branch
+              git fetch origin "$TARGET_BRANCH"
+              git checkout -b "$CHERRY_PICK_BRANCH" "origin/$TARGET_BRANCH"
+
+              # Attempt cherry-pick
+              if git cherry-pick "$COMMIT_SHA"; then
+                echo "✅ Cherry-pick successful for $TARGET_BRANCH"
+
+                # Push the cherry-pick branch
+                git push origin "$CHERRY_PICK_BRANCH"
+
+                # Create PR for the cherry-pick
+                CHERRY_PICK_TITLE="$PR_TITLE (backport of #$PR_NUMBER)"
```

```suggestion
                CHERRY_PICK_TITLE="$PR_TITLE (cherry-pick #$PR_NUMBER)"
```
Not necessary, only for consistency with past PRs